pkg/plugins/execplugin.go

```diff
@@ -202,3 +195,43 @@ func (p *ExecPlugin) getEnv() []string {
 		"KUSTOMIZE_PLUGIN_CONFIG_ROOT="+p.ldr.Root())
 	return env
 }
+
+func (p *ExecPlugin) getResMapWithIdAnnotation(rm resmap.ResMap) resmap.ResMap {
+	inputRM := rm.DeepCopy(p.rf.RF())
+	for id, r := range inputRM {
+		annotations := r.GetAnnotations()
+		if annotations == nil {
+			annotations = make(map[string]string)
+		}
+		annotations[idAnnotation] = id.String()
+		r.SetAnnotations(annotations)
+	}
+	return inputRM
+}
+
+func (p *ExecPlugin) parseResMapFromOutput(output []byte, rm resmap.ResMap) error {
+	outputRM, err := p.rf.NewResMapFromBytes(output)
+	if err != nil {
+		return err
+	}
+	for _, r := range outputRM {
+		annotations := r.GetAnnotations()
```

Comment:
`// for each emitted Resource, find the matching Resource in the original resMap using its id`